include/behaviortree_cpp/basic_types.h

```diff
@@ -450,6 +452,43 @@ struct has_static_method_description<
 {
 };
 
+/// Optional metadata for a TreeNodeManifest.
+/// The metadata can represent an xml element with text:
+///   <metadata_name>Descriptive text</metadata_name>
+/// Or an xml element with an attribute:
+///   <metadata_name attribute="value"/>
```

Comment:
Do we also want to support the following case?

```
 <metadata_name attribute1="value1" attribute2="value2"/>
```

If so, might be worth the `std::variant` in question to use `std::vector<Attribute>` rather than (or in addition to) `Attribute` and it all may just shake out.

Reply:
Sounds like we don't need the variant anymore once I implement the approach Davide asked for in his review.